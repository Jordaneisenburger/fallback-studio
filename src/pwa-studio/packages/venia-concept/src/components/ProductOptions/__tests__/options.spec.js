import React from 'react';
import testRenderer from 'react-test-renderer';

import Options from '../options';

jest.mock('../../../classify');
jest.mock('uuid/v4', () => () => '00000000-0000-0000-0000-000000000000');

const onSelectionChangeMock = jest.fn();
const defaultProps = {
    product: {
        __typename: 'ConfigurableProduct',
        configurable_options: [
            {
                attribute_id: '1',
                attribute_code: 'fashion_color',
                label: 'option-1',
                values: []
            },
            {
                attribute_id: '2',
                attribute_code: '',
                label: 'option-1',
                values: []
            }
        ]
    }
};
test('renders Options component correctly', () => {
    const component = testRenderer.create(<Options {...defaultProps} />);
    expect(component.toJSON()).toMatchSnapshot();
});

test('renders null if product is not configurable', () => {
    const simpleProduct = {};
    const component = testRenderer.create(<Options product={simpleProduct} />);
    expect(component.toJSON()).toMatchSnapshot();
});

test('does not call onSelectionChange if not provided', () => {
    const component = testRenderer.create(<Options {...defaultProps} />);
    component.getInstance().handleSelectionChange(0, 'test');
    expect(onSelectionChangeMock).not.toHaveBeenCalled();

    onSelectionChangeMock.mockReset();
});

test('calls onSelectionChange function with optionId and selection', () => {
    const props = {
        ...defaultProps,
        onSelectionChange: onSelectionChangeMock
    };
    const component = testRenderer.create(<Options {...props} />);

    component.getInstance().handleSelectionChange(0, 'test');
    expect(onSelectionChangeMock).toHaveBeenCalledWith(0, 'test');

    onSelectionChangeMock.mockReset();
});
