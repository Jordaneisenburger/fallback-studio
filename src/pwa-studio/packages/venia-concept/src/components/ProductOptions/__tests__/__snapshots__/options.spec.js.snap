// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Options component correctly 1`] = `
Array [
  <div
    className="root"
  >
    <h3
      className="title"
    >
      <span>
        option-1
      </span>
    </h3>
    <div
      className="root"
    />
  </div>,
  <div
    className="root"
  >
    <h3
      className="title"
    >
      <span>
        option-1
      </span>
    </h3>
    <div
      className="root"
    />
  </div>,
]
`;

exports[`renders null if product is not configurable 1`] = `null`;
